toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -110,13 +119,85 @@ func fetchPackages() (err error) {
 		return
 	}
 
+	// Optional delta build cache hydration
```

Comment:
@PawelWMS 
> Now when I see both the refactoring and these changes: why do we want to save the graph even if we may need to overwrite it later?

This comment got orphaned.

Reply:
Simple answer is convenience, the scheduler code all expects to read its input from disk. I didn't want to start refactoring that code as well.

We can revisit later maybe, and split up the scheduler code into reading from disk + optimizing.

Reply:
Was able to refactor